File: ai-service/routes/schemes.py
from fastapi import APIRouter
from pydantic import BaseModel
from typing import List, Optional

router = APIRouter()

class SchemeRequest(BaseModel):
    age: int
    gender: str
    income: Optional[int] = None
    conditions: Optional[List[str]] = []
    category: Optional[str] = None
    state: Optional[str] = None

SCHEMES = [
    {"id": "pmjay", "name": "Ayushman Bharat PM-JAY", "coverage": 500000,
     "eligibility": {"max_income": 100000, "categories": ["BPL", "SC", "ST"]}},
    {"id": "jssk", "name": "Janani Shishu Suraksha", "coverage": None,
     "eligibility": {"gender": "female", "age_range": [15, 45]}},
    {"id": "rbsk", "name": "Rashtriya Bal Swasthya", "coverage": None,
     "eligibility": {"age_range": [0, 18]}},
    {"id": "npcdcs", "name": "NPCDCS", "coverage": None,
     "eligibility": {"conditions": ["diabetes", "hypertension", "cancer"]}},
]

@router.post("/schemes")
def check_schemes(req: SchemeRequest):
    eligible = []
    for scheme in SCHEMES:
        e = scheme["eligibility"]
        if "max_income" in e and req.income and req.income > e["max_income"]:
            continue
        if "gender" in e and e["gender"] != req.gender:
            continue
        if "age_range" in e and not (e["age_range"][0] <= req.age <= e["age_range"][1]):
            continue
        if "categories" in e and req.category not in e["categories"]:
            continue
        if "conditions" in e and not any(c in req.conditions for c in e["conditions"]):
            continue
        eligible.append(scheme)
    return {"eligible_schemes": eligible, "count": len(eligible)}

File: backend/src/routes/schemes.js
const router = require('express').Router();
const { protect } = require('../middleware/auth');

const SCHEMES = [
  { id: 'pmjay', name: 'Ayushman Bharat PM-JAY', description: 'Health coverage up to ₹5 lakh per family', eligibility: { maxIncome: 100000, categories: ['BPL', 'SC', 'ST'] } },
  { id: 'jssk', name: 'Janani Shishu Suraksha Karyakram', description: 'Free maternity services', eligibility: { gender: 'female', ageRange: [15, 45] } },
  { id: 'rbsk', name: 'Rashtriya Bal Swasthya Karyakram', description: 'Child health screening', eligibility: { ageRange: [0, 18] } },
  { id: 'npcdcs', name: 'NPCDCS', description: 'Non-communicable disease control', eligibility: { conditions: ['diabetes', 'hypertension', 'cancer'] } }
];

router.use(protect);

router.get('/', (req, res) => res.json(SCHEMES));

router.post('/check-eligibility', (req, res) => {
  const { age, gender, income, conditions = [], category } = req.body;
  const eligible = SCHEMES.filter(scheme => {
    const e = scheme.eligibility;
    if (e.maxIncome && income > e.maxIncome) return false;
    if (e.gender && e.gender !== gender) return false;
    if (e.ageRange && (age < e.ageRange[0] || age > e.ageRange[1])) return false;
    if (e.categories && !e.categories.includes(category)) return false;
    if (e.conditions && !e.conditions.some(c => conditions.includes(c))) return false;
    return true;
  });
  res.json(eligible);
});

module.exports = router;
